Online version: Use CodeMirror in the grammar editor

djadmin opened the pull request:
As discussed here https://github.com/pegjs/website/issues/12#issuecomment-94579536, I'm requesting you a pull request so that you can review my code and the feature so that it can be embedded in the pegjs website.

Please let me know if I've missed anything.

Regards
Dheeraj

dmajda commented:
@djadmin Thanks for your work, I just had a quick look at the demo [you mentioned](https://github.com/pegjs/website/issues/12#issuecomment-94579536) and it looks great.

Could you please update your code so it correctly applies to the current master? I’d have a look then and do a more detailed review.

And sorry I got to this a bit late.

djadmin commented:
Hi,
Sorry for the delay. I have updated the codebase and fixed the issues as well.

Please review my work and let me know if I need to make any changes. Would love to do improvements.

Regards
Dheeraj

dmajda commented:
@djadmin Thanks. I’m focusing on the [0.9.0 release](https://github.com/pegjs/pegjs/issues/364) now, I’ll have a look once it’s out.

dmajda commented:
@djadmin Could you please rebase the branch so that is doesn’t include my commits and squash your changes into one commit? This will allow me to apply your patch cleanly, try it out, and add meaningful review comments. Thanks.

djadmin commented:
Hi, I have updated my code to latest CodeMirror and squashed my commits into one. Now it's ready to be reviewed.
Thanks

dmajda commented:
Final thoughts:

  * It would be god to use CodeMirror also for the output for consistency. But this is not a requirement for the first version.
  * Visually, the default theme doesn’t fit PEG.js website very well (mainly because of used colors). It would be good to find a theme that fits better or write/adapt one. Again, not a requirement for the first version.

djadmin commented on public/css/content.css:
```diff
@@ -13,7 +13,7 @@
   background-color: #f0f0f0;
 }
-#content table { border-spacing: 0; }
+#content table { border-spacing: 0; table-layout: fixed;}
```
If it's not set, CodeMirror expands horizontally if you exceed the content instead of showing scrollbar. Verified.

djadmin commented:
@dmajda Thank for the feedback.
Yeah agree, we could use it for the output as well and as you mentioned before we can show errors in gutters.

On the list:
1) Select appropriate theme
2) Make a build system for handling vendor libraries.
3) Errors message with codemirror
4) Autocomplete

djadmin commented:
@dmajda Hi, The bug you mentioned here https://github.com/djadmin/website/commit/bc149c021c20afdc25876aab0a09e9baf1925543#commitcomment-13322268 has been resolved. 
It was causing issue because We have been meddling with the dom and codemirror needs to be refreshed after that to resolve such issues.

dmajda commented:
I looked at the latest version, it looks great! I still have a couple of minor comments which I’ll add shortly. They should be really easy to address.

One global thing: Could you update the commit message to be correct content-, grammar-, and spelling-wise? Ideally something like `Online version: Use CodeMirror in the grammar editor`.

djadmin commented:
Hi,

I've addressed all of your comments and updated the same. Let me know if I have missed out anything. You can access and verify the changes on a test app (https://pegjs-website.herokuapp.com/online).

Thank you

dmajda commented:
Everything looks great now, I will merge and deploy the website shortly. Many thanks for the work and for the patience with me!
